src/ds/redblacktree.h

```diff
@@ -41,7 +65,20 @@ namespace snmalloc
     {
       Rep::ref(b, k)
     }
-    ->ConceptSame<typename Rep::Holder&>;
+    ->ConceptSame<typename Rep::Holder>;
+    {
+      Rep::null
+    }
+    ->ConceptSameModRef<const typename Rep::Contents>;
+    {
+      typename Rep::Holder
+      {
+        const_cast<
+          std::remove_const_t<std::remove_reference_t<decltype(Rep::root)>>*>(
```

Comment:
I see why you need to remove `const` but not reference types.

Reply:
If `Rep` declares a static `root` of type `T` then `decltype(Rep::root)` will be `T&`.